### PR TITLE
[RFR] Update CatalogItem callers to pass a provider object

### PR DESCRIPTION
**Pull request by mshriver**
{{pytest: cfme/tests/services/ --long-running}}

PR 3894 updated CatalogItem class to take a provider object instead of the name.

Update callers to pass the object

PRT Results:

Lots of failures from form filling and flash message assertions, don't appear to be related to my changes but I'm debugging and trying to clean up the tests.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 03953b15a5377033c2ca4069c6daf9901b79ab4e:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit faeda5610ec7e2477698eef5d0b9b5dfc9b3b9d4:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit aeb58a4a48cf08f34ee0c5b8ea7620a64f74ee23:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5bfce3189cc35e61cdf2a1a14861193a81c7440e:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit aa938a401a14812c4b7365471a5f613b47ba7ad2:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6e60d8b20b0f15d099f47b3622c81438c184fa83:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 73ab76e1f991db2db0da2ebf96ff37b2c8170ede:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/services/catalogs/catalog_item.py [hunk 1]
```diff
@@ -145,17 +131,18 @@ def create(self):
...
         catalog = fakeobject_or_object(self.catalog, "name", "Unassigned")
         dialog = fakeobject_or_object(self.dialog, "name", "No Dialog")
-        sel_provider = None
-        if self.item_type is not "Generic":
-            if self.provider.type_tclass is not "infra":
-                sel_provider = self.provider
+        # Provider is only in the create form for AnsibleTower type
+        select_provider = None
+        if not isinstance(self.provider, NoneType) \
+                and 'AnsbileTower' in [self.item_type, self.provider_type]:
```
Comment (mshriver) [inline cfme/services/catalogs/catalog_item.py]: @sshveta can you check this out? I went through the 5.6 and 5.7 catalog item create forms, and in both cases the provider selection was _only_ present when Ansible Tower was selected as the type in the first part of the form.

The logic you had put in before was checking provider.type_tclass was not infra? What was the goal/intent there, what behavior in the UI were you trying to model? Thanks!

Comment (sshveta) [inline cfme/services/catalogs/catalog_item.py]: @mshriver : The provider field needs to be selected for orchestration provisioning (ec2 and heat ) and ansible , that means cloud and ansible , so filtering out infra and generic types.

**Comment by dajoRH** [comment]
Lint report for commit c6a7dbf030c1e6dbaecffb1a2b8d69b023c5af3f:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 13bde9bb0651cea83b4a574c676772f6a24c30e7:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by psav** [comment]
Failure analysis?

**Comment by dajoRH** [comment]
Lint report for commit 729eb5887895623860d12ff224a69a33b97fcefe:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**